engine/runtime-language-arrow/src/main/java/org/enso/interpreter/arrow/runtime/ByteBufferDirect.java

```diff
@@ -0,0 +1,185 @@
+package org.enso.interpreter.arrow.runtime;
+
+import com.oracle.truffle.api.interop.UnsupportedMessageException;
+import java.nio.ByteBuffer;
+
+final class ByteBufferDirect extends ByteBufferProxy {
+  private final ByteBuffer buffer;
+  private final ByteBuffer nonNullBitmap;
```

Comment:
Doesn't Arrow specification require the `nonNullBitMap` and `buffer` to follow each other? In such case you'd have to have just a single `ByteBuffer` for both...

Reply:
I don't see that being specified anywhere

Reply:
OK, discussed this offline. Seems like specification implies this indirectly. I will leave it in this PR but at it to the list of improvements to the follow up.